weave/flow/eval_imperative.py

```diff
@@ -389,12 +389,18 @@ def model_post_init(self, __context: Any) -> None:
         # objects that "look right" to our object saving system.
 
         # --- Setup the model object ---
-        @weave.op(name="Model.predict", enable_code_capture=False)
-        def predict(self: Model, inputs: dict) -> Any:
-            # Get the output from the context variable
-            return current_output.get()
+        # Only modify the predict method if the model doesn't already have one
+        try:
+            assert isinstance(self.model, Model)
```

Comment:
I think this is going to raise an `AssertionError` that will not be caught by your `except` clause.